code/test_range_overlap.m

```diff
@@ -1,9 +1,9 @@
 %script test_range_overlap.m
 
 
-% assert(range_overlap([0.0, 1.0], [5.0, 6.0]) == NaN);
-% assert(range_overlap([0.0, 1.0], [1.0, 2.0]) == NaN);
-assert(range_overlap([0, 1.0]) == [0, 1.0]);
-assert(range_overlap([2.0, 3.0], [2.0, 4.0]) == [2.0, 3.0]);
-assert(range_overlap([0.0, 1.0], [0.0, 2.0], [-1.0, 1.0]) == [0.0, 1.0]);
+assert(isnan(range_overlap([0.0, 1.0], [5.0, 6.0]))==1);
+assert(isnan(range_overlap([0.0, 1.0], [1.0, 2.0]))==1);
+assert(isequal(range_overlap([0, 1.0]),[0, 1.0]));
+assert(isequal(range_overlap([2.0, 3.0], [2.0, 4.0]),[2.0, 3.0]));
+assert(isequal(range_overlap([0.0, 1.0], [0.0, 2.0], [-1.0, 1.0]),[0.0, 1.0]));
```

Comment:
Some introduction to functions like `isequal` and `isnan` (as opposed to `==` and `== Nan`) is required here.